docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,293 @@
+
+This guide is focused on apps which use OAuth to make authorized ("authz") requests to user PDS servers, for example to write records to atproto repositories, or make proxied API requests to other services. It is also possible to use the atproto identity system only for authentication ("authn"), similar to OpenID/OIDC.
+
+## Types of Clients
+
+This guide covers three simplified types of OAuth client:
+
+- **Web Services**: traditional web apps which involve a server/backend running code to make actual PDS requests and talk with a database. There is some form of auth between browsers and the web service, such as cookie sessions; this auth layer is distinct from OAuth. The server may return complete HTML pages, or there may be an API between code running in the browser and code running on the server. Integrations with and extensions of existing web services also fall under this category.
+- **Browser Apps**: "single-page" applications which run in a web browser, implemented using web platform APIs and JavaScript or WASM runtimes. The server-side ("backend") component is minimal, or even just static file hosting.
+- **Mobile and Desktop Apps**: what they sound like: "native" apps that run on mobile operating systems (smartphones, tablets, etc), or desktop applications
+
+|                         | **Web Service** | **Browser App** | **Mobile or Desktop App** |
+| ---                     | ---             | ---             | ---                       |
+| **OAuth 2 Client Type** | "Confidential"  | "Public"        | "Public"                  |
+| `client_id` | ✅ URL to metadata | ✅ URL to metadata | ✅ URL to metadata |
+| `client_secret` | ❌  | ❌  | ❌  |
+| **OAuth 2 Grant Types** | `authorization_code`
+`refresh_token` | `authorization_code`
+`refresh_token` | `authorization_code`
+`refresh_token` |
+| **Client Metadata** | ✅ Public Web | ✅ Public Web | ✅ Public Web |
+| **Client Metadata JWK** | ✅ Public Web | ❌  | ❌  |
+| **PKCE** | ✅ | ✅ | ✅ |
+| **PAR** | ✅ | ✅ | ✅ |
+| **DPoP** | ✅ | ✅ | ✅ |
+| **Handle Resolution** | DNS and HTTPS | via helper service | DNS and HTTPS or via helper service |
+| **DID Resolution** | HTTPS | HTTPS | HTTPS |
+| **Recommended Client Secret Key Storage** | Environment Variable, Secrets Manager, Hardware Enclave | ❌  | ❌  |
+| **Recommended DPoP Key Storage** | Secure Database | non-exportable CryptoKeyPair in IndexedDB | Secure File or Database, Hardware Enclave |
+| **Recommended Token Storage** | Secure Database | IndexedDB or LocalStorage | Secure File or Database |
+| **SSRF + DoS Hardening** | ✅ | ✅ | ✅ |
+| **Authorization UI** | Browser Redirect  | Browser Redirect  | WebView/Browser |
+| `redirect_uri` | `https://` App URL | `https://` App URL | Client-specific URI scheme |
```

Comment:
May want a double-check, but this seems fairly uncontroversial to me:
```suggestion
| `redirect_uri` | `https://` App URL | `https://` App URL | App Link (Android), Universal Link (iOS), or Client-specific URI scheme |
```

https://developer.android.com/training/app-links
https://developer.apple.com/ios/universal-links/
https://bsky.app/profile/krasovs.ky/post/3l2tb4jcq4s22